File: proxy/tests/script_engine/query_delete.js
/// <reference path="../../script_libs/js/roxy.d.ts" />
/** @type {Extension} */
const query_delete = {
  request(flow) {
    if (flow.request.url.searchParams.get("foo") == "bar") {
      flow.request.url.searchParams.delete("foo")
    }
  }
};
globalThis.extensions = [query_delete];

File: proxy/tests/script_engine/query_delete.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local query_delete = {
	request = function(flow)
		if flow.request.url.search_params["foo"] == "bar" then
			flow.request.url.search_params:delete("foo")
		end
	end,
}
Extensions = { query_delete }
